document/document.go

```diff
@@ -29,11 +30,16 @@ func init() {
 	reflectStaticSizeDocument = int(reflect.TypeOf(d).Size())
 }
 
+const SynonymDocument = true
+const NormalDocument = false
```

Comment:
don't think this var is needed from what i see it says that the document is not synonymEnabled, so in the line 54, its better to do just say DocType: !SynonymDocument